graph/src/util/backoff.rs

```diff
@@ -33,7 +33,7 @@ impl ExponentialBackoff {
     }
 
     pub fn delay(&self) -> Duration {
-        let mut delay = self.base.saturating_mul(1 << self.attempt);
+        let mut delay = self.base.saturating_mul(1u32 << self.attempt.min(31));
```

Comment:
I think it would be better to make sure that `next_attempt` never increments `self.attempt` to more than 31. We can also do the `min` here in addition, but I think we should also clamp `self.attempt` to 31 explicitly

Reply:
Thought it would be better to let it increment to have a sense how many total attempts there were, i.e. for logging.

Reply:
True. That makes sense